contracts/misc/ERC721AtomicOracleWrapper.sol

```diff
@@ -0,0 +1,159 @@
+    /**
+     * @dev This constant represents the minimum price multiplier that a single tokenId can have
+     * A value of 1e18 results in no price multiplier
+     */
+    uint256 internal constant MIN_PRICE_MULTIPLIER = 1e18;
+
+    /**
+     * @notice NFT type multipliers
+     * nftType <-> priceMultiplier
+     */
+    mapping(bytes32 => uint256) public nftTypeMultipliers;
+
+    /**
+     * @notice NFT types
+     * tokenId <-> nftType
+     */
+    mapping(uint256 => bytes32) public nftTypes;
+
+    /**
+     * @dev Only asset listing or pool admin can call functions marked by this modifier.
+     **/
+    modifier onlyAssetListingOrPoolAdmins() {
+        _onlyAssetListingOrPoolAdmins();
+        _;
+    }
+
+    function _onlyAssetListingOrPoolAdmins() internal view {
+        IACLManager aclManager = IACLManager(
+            ADDRESSES_PROVIDER.getACLManager()
+        );
+        require(
+            aclManager.isAssetListingAdmin(msg.sender) ||
+                aclManager.isPoolAdmin(msg.sender),
+            Errors.CALLER_NOT_ASSET_LISTING_OR_POOL_ADMIN
+        );
+    }
+
+    constructor(
+        address _provider,
+        address _aggregator,
+        address _asset
+    ) {
+        ADDRESSES_PROVIDER = IPoolAddressesProvider(_provider);
+        aggregator = IEACAggregatorProxy(_aggregator);
+        asset = _asset;
+    }
+
+    function setOracle(address _aggregator)
+        external
+        onlyAssetListingOrPoolAdmins
+    {
+        aggregator = IEACAggregatorProxy(_aggregator);
+    }
+
+    function _validatePriceMultiplier(uint256 _priceMultiplier) internal pure {
+        require(
+            _priceMultiplier >= MIN_PRICE_MULTIPLIER &&
+                _priceMultiplier <= MAX_PRICE_MULTIPLIER,
+            "invalid price multiplier"
+        );
+    }
+
+    /**
+     * @notice Set price multiplier for the specified tokenId;
+     */
+    function setPriceMultiplier(bytes32 nftType, uint256 _priceMultiplier)
+        external
+        onlyAssetListingOrPoolAdmins
+    {
+        _validatePriceMultiplier(_priceMultiplier);
+        require(nftType != bytes32(0), "invalid nftType");
+        nftTypeMultipliers[nftType] = _priceMultiplier;
+    }
+
+    /**
+     * @notice Set NFT type for a group of tokens
+     */
+    function setNFTType(bytes32 nftType, uint256[] calldata tokenIds)
+        external
+        onlyAssetListingOrPoolAdmins
+    {
+        require(nftType != bytes32(0), "invalid nftType");
+
+        for (uint256 i; i < tokenIds.length; ++i) {
+            nftTypes[tokenIds[i]] = nftType;
+        }
+    }
+
+    /**
+     * @notice Returns the multiplied price for the specified tokenId.
+     */
+    function getTokenPrice(uint256 tokenId) public view returns (uint256) {
+        bytes32 nftType = nftTypes[tokenId];
+        uint256 multiplier = nftTypeMultipliers[nftType];
+        uint256 price = uint256(aggregator.latestAnswer());
+        if (nftType != bytes32(0) && multiplier > 0) {
```

Comment:
should't `multiplier > 0` be enough? since `nftType == bytes32(0)` should always result in `multiplier == 0`?